conans/client/rest/rest_client_v2.py

```diff
@@ -206,15 +221,39 @@ def _download_and_save_files(self, base_url, dest_folder, files):
             downloader.download(resource_url, abs_path, auth=self.auth)
 
     def _recipe_url(self, conan_reference):
-        url = "%s/conans/%s" % (self.remote_api_url, "/".join(conan_reference))
+
+        url = "%s/conans/%s" % (self.remote_api_url, conan_reference.dir_repr())
 
         if conan_reference.revision:
-            url += "#%s" % conan_reference.revision
-        return url.replace("#", "%23")
+            url += "/revisions/%s" % conan_reference.revision
+        return url
 
     def _package_url(self, p_reference):
+        if not p_reference.conan.revision and p_reference.revision:
+            raise ConanException("It is needed to specify the recipe revision if you "
+                                 "specify a package revision")
+
         url = self._recipe_url(p_reference.conan)
         url += "/packages/%s" % p_reference.package_id
         if p_reference.revision:
-            url += "#%s" % p_reference.revision
-        return url.replace("#", "%23")
+            assert(p_reference.revision is not None)
+            url += "/revisions/%s" % p_reference.revision
+        return url
+
+    def _remove_conanfile_files(self, conan_reference, files):
+        # V2 === revisions, do not remove files, it will create a new revision if the files changed
```

Comment:
A bit weird, empty functionality being called. It seems that intent would be more clear if the caller didn't call it, it already knows that a new revision will be created?

Can be left as-is, though, if anything lets review later.